exercises/sieve/example.rs

```diff
@@ -0,0 +1,33 @@
+pub struct Sieve { limit: usize }
+
+impl Sieve {
+    pub fn new(limit: usize) -> Sieve {
+        Sieve { limit: limit }
+    }
+
+    pub fn primes(&self) -> Vec<usize> {
+        let mut primes = (0..self.limit).map ( |x| x + 1).collect::<Vec<usize>>();
+        let tracker = primes.clone();
+
+        for num in tracker {
+            if !prime(num) {
+                if num > 1 {
+                    primes.retain( |&x| x % num != 0);
+                } else {
+                    primes.retain( |&x| x > 1);
+                }
+            }
+        }
+
+        primes
+    }
+
+}
+
+pub fn prime(num: usize) -> bool {
```

Comment:
do we care very much about a very strict interpretation of the sieve algorithm's operation? If we did, we'd want to check primality via whether `num` hasn't already been marked off (whether it's still in `primes`, perhaps?), rather than re-iterating through all numbers less than `num`